### PR TITLE
new trait MessagesInjectedController

### PR DESCRIPTION
**Pull request by hsn10**

### CODE REVIEW

File: framework/src/play/src/main/scala/play/api/mvc/Controller.scala [hunk 1]
```diff
@@ -194,6 +194,34 @@ trait InjectedController extends BaseController {
   }
 }
 
+/**
+ * A variation of [[MessagesAbstractController]] that gets its components via method injection.
+ */
+trait MessagesInjectedController extends BaseController {
```
Comment (gmethvin) [inline framework/src/play/src/main/scala/play/api/mvc/Controller.scala]: Shouldn't this extend `MessagesBaseController`? As-is you wouldn't get the custom definition for `Action: ActionBuilder[MessagesRequest, AnyContent]`

**Comment by gmethvin** [comment]
I'm personally not a fan of the method injection style, but I suppose since we're already providing `InjectedController` it makes sense to provide this as well. 🤷‍♂️

File: framework/src/play/src/main/scala/play/api/mvc/Controller.scala (hunk 1 above)
Comment (hsn10) [reply framework/src/play/src/main/scala/play/api/mvc/Controller.scala]: right

**Comment by marcospereira** [comment]
Backport to 2.7.x: cfdd7e4d8d8a3f2ec20281a07bec508719e5e332

**Pull request merged**